app/uk/gov/hmrc/helptosavefrontend/models/HTSEvent.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavefrontend.models
+
+import uk.gov.hmrc.play.audit.model.DataEvent
+import uk.gov.hmrc.play.http.HeaderCarrier
+import uk.gov.hmrc.play.audit.AuditExtensions._
+
+abstract class HTSEvent(auditType: String, detail: Map[String,String])(implicit hc: HeaderCarrier)
+  extends DataEvent(auditSource = "hts-frontend", auditType = auditType, detail = detail, tags = hc.toAuditTags("", "N/A"))
+
+class ApplicationSubmittedEvent(userInfo: NSIUserInfo)(implicit hc: HeaderCarrier)
+  extends HTSEvent("applicationSubmitted",
+    Map[String, String](
+        "forename" -> userInfo.forename,
+        "surname" -> userInfo.surname,
+        "dateOfBirth" -> userInfo.dateOfBirth.toString,
+        "nino" -> userInfo.nino,
+        "address1" -> userInfo.contactDetails.address1,
+        "address2" -> userInfo.contactDetails.address2,
+        "address3" -> {userInfo.contactDetails.address3.fold("") {identity}},
+        "address4" -> {userInfo.contactDetails.address4.fold("") {identity}},
+        "address5" -> {userInfo.contactDetails.address5.fold("") {identity}},
+        "postcode" -> userInfo.contactDetails.postcode,
+        "countryCode" -> {userInfo.contactDetails.countryCode.fold("") {identity}},
+        "email" -> userInfo.contactDetails.email,
+        "phoneNumber" -> {userInfo.contactDetails.phoneNumber.fold(""){identity}},
+        "communicationPreference" -> userInfo.contactDetails.communicationPreference,
+        "registrationChannel" -> userInfo.registrationChannel))
```

Comment:
ok, i thought we are not supposed to log customer's private information like dob.. etc..